bach/bach/series/series.py

```diff
@@ -340,8 +326,15 @@ def get_class_instance(
 
     @classmethod
     def get_db_dtype(cls, dialect: Dialect) -> str:
-        """ Given the db_dtype of this Series, for the given database dialect. """
+        """
+        Give the static db_dtype of this Series, for the given database dialect.
+        :raises DatabaseNotSupportedException:  If the db_dtype is not defined for the given dialect.
+        """
         db_dialect = DBDialect.from_dialect(dialect)
+        if db_dialect not in cls.supported_db_dtype:
+            raise DatabaseNotSupportedException(
+                dialect,
+                message_override=f'Cannot get db type of {cls.name} for {dialect}')
```

Comment:
`cls.name` is actually a property for the series name.
```suggestion
                message_override=f'Cannot get db type of {cls.__name__} for {dialect.name}')
```

Reply:
doh. you are right of course